compiler/builtins/string.ts

```diff
@@ -1275,7 +1275,43 @@ i32.store8 0 12`;
       Porffor.wasm.i32.store16(Porffor.wasm`local.get ${tmp}` + tmpLen * 2, x, 0, 4);
       tmpLen++;
     }
-  } else {
+  } 
+  else if (sepLen === 0) {
+    const clammedLimit: i32 = limit > thisLen ? thisLen : limit;
```

Comment:
This is great but unfortunately overcomplicated, as strings are immutable they don't need complex cloning like this. Instead you should be able to just do something like
```ts
out.length = 1;
out[0] = _this;
return out;
```